tools/ccm_extend.py

```diff
@@ -17,10 +18,28 @@
 headers = {'Authorization': 'Token {}'.format(ccm_token)}
 
 # Determine the cluster id of the cluster
-all_clusters = json.loads(requests.get(base_url+'/api/cluster/active/all', headers=headers).text)
-for cluster in all_clusters:
-    if cluster['name'] == cluster_name:
-        cluster_id = cluster['id']
+# It may take some time for CCM to have all of the clusters information. Poll
+# until it receives it. Try for 60 minutes, because we really don't want to waste a cluster we just created.
+give_up_time = time.time() + 60 * 60
+
+while give_up_time > time.time():
+    print("Looking up cluster id.")
+    # Get all the clusters
+    all_clusters = json.loads(requests.get(base_url+'/api/cluster/active/all', headers=headers).text)
+    for cluster in all_clusters:
+        if cluster['name'] == cluster_name:
+            cluster_id = cluster['id']
+
+    if cluster_id is not None:
+        print("Cluster id found!")
+        break
+
+    print("Cluster id not found. Sleeping for 60 seconds...")
+    time.sleep(60)
+
+if cluster_id is None:
+    print("Never retrieved cluster id from CCM!")
+    os.exit(1)
 
 # This API call increases the cluster length by an additional 4 hours.
 # See: https://ccm.mesosphere.com/api-docs/#!/cluster/Cluster_Detail_PUT
```

Comment:
On the line below, the spaces on either side of the `=` are not pythonic for specifying keyword arguments and the `data` dictionary is missing a space after the `:`.